lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -464,16 +464,37 @@ struct GetNumProgramsOpConversion
     Location loc = op->getLoc();
     assert(op.getAxis() < 3);
 
+#ifdef USE_ROCM
+    // Seem like GridDimOp returns the number of threads (not the number of 
+    // workgroups) in a kernel (a bug in llvm https://reviews.llvm.org/D156009), 
+    // so as a workaround here, we divide by the number of threads 
+    // per workgroup to get the number of workgroups in a kernel.
+    // TODO: when we do upstream to include llvm fix, we can remove this workaround
+    // The unit test added in this PR can guarantee that.
+    Value threadsPerGrid =
+        rewriter.create<::mlir::gpu::GridDimOp>(loc, dims[op.getAxis()]);
+    Value threadsPerBlock =
+        rewriter.create<::mlir::gpu::BlockDimOp>(loc, bdims[op.getAxis()]);
```

Comment:
Why cannot we reuse `dims` in this case?

Reply:
My understanding is that dims is used to store the `gridDim` information, so we need another one to store the blockDim information.

Reply:
`dims` is just a const array providing the three dimensions. And you do want `threadsPerGrid` and `threadsPerBlock` to have the same dim. So I think you can safely reuse `dims` and remove `bdims` below.

Reply:
Thanks. Fixed.